### ISSUE TITLE
CRC error when sensor description > 25 chars

### ISSUE THREAD

**Issue by MaartenSp** [OpenLightingProject/ola#1551]
I have the following situation:

- RDM device connected, containing three sensors
- View the sensor values via the OLA web interface

In this case, when one of the sensor description contains more than 25 characters, the description is not displayed. When I run the rdm tests and run GetSensorDefinition there, I see an invalid CRC error for the sensors containing more than 25 characters in the description. The other ones are fine. According to RDM standards, a sensor description should be able to contain 32 characters.

I checked with other controllers, which also check CRC, and they work fine for the case mentioned before.

**Comment by peternewman** [comment]
Hi @MaartenSp ,

Can we have some logs please, both olad -l 4 ( https://www.openlighting.org/ola/get-help/ola-faq/#How_do_I_get_olad_-l_4_logs ) and if you run just that RDM test via the rdm_responder_test.py with debug turned on (i.e. using -d and -t):
http://docs.openlighting.org/ola/man/man1/rdm_responder_test.py.1.html

What dongle/interface are you using? What's the responder (is it one you've written yourself)?

Can you also try using ola_rdm_get with --frames and -l 4:
http://docs.openlighting.org/ola/man/man1/ola_rdm_get.1.html

Which version of OLA, which OS etc?

**Comment by MaartenSp** [comment]
[oladLog.txt](https://github.com/OpenLightingProject/ola/files/3022624/oladLog.txt)
[rdm_responder_test_log.txt](https://github.com/OpenLightingProject/ola/files/3022625/rdm_responder_test_log.txt)
[ola_rdm_get_log.txt](https://github.com/OpenLightingProject/ola/files/3022627/ola_rdm_get_log.txt)

Hi @peternewman ,

Thanks for your reply. I added the following:

- Olad log file
- rmd_responder_test output
- ola_rdm_get output

We are running OLA version 10.7, on Raspbian Stretch Light, kernel version 4.14. The responder is one we're currently developing. We are using an ENTTEC DMXUSB PRO interface.

In the ola_rdm_get_log file I added the faulty case (26 chars) and a passing case (25 characters). It seems in both cases the crc in the raw RDM frames is correct (the sum of all data values).

Please let me know if you need any additional info.

**Comment by MaartenSp** [comment]
The raw RDM frame output is really helpful. I went through the frames again, and I do see a problem in the CRC, I must have missed that the first time. Thanks for pointing me in the right direction!

**Comment by MaartenSp** [comment]
The third sensor gives this output:

pi@raspberrypi:~ $ ola_rdm_get --frames  --universe 0 --uid 4658:0090000e sensor_definition 2
Error: Incorrect checksum
------- Frame Information --------
cc 01 40 45 4e 02 14 99  ..@EN...
40 46 58 00 90 00 0e 47  @FX....G
00 00 00 00 21 02 00 28  ....!..(
02 7f 00 00 00 00 00 64  .......d
00 00 00 64 00 31 32 33  ...d.123
34 35 36 37 38 39 30 31  45678901
32 33 34 35 36 37 38 39  23456789
30 31 32 33 34 35 36 37  01234567
00 2c                    .,

The odd thing is that the 65th byte is displayed as 00 in both faulty cases.

**Comment by MaartenSp** [comment]
Thank you @peternewman. The command to display the raw RDM frames made debugging a lot easier and pointed me in the right direction, The problem was in the responder and is solved now. This issue may be closed. Thanks for your help!

**Comment by peternewman** [comment]
Excellent, glad you've sorted it @MaartenSp and that you found the raw frames helpful.

For future reference, would you care to tell us the "other controllers, which also check CRC, and they work fine for the case mentioned before.", just so we can be aware they're perhaps not as thorough as expected.

In terms of developing fixtures, you might also find this useful, it does the maths for you, (and the other bits under the tools menu):
http://rdm.openlighting.org/tools/packet-extractor

**Comment by MaartenSp** [comment]
The problem did not always reproduce, even with the ola_rdm_get command I got different results in each run. The controller I used most was the NA MasterFade, which did not show the problem, but that is not a 100% guarantee that the controller is less thorough. I also used a Swisson XMT-350, which worked fine.

Thanks for the link to the RDM packet dissector, it's a convenient tool and could be useful in the future :)
